### PR TITLE
2236 Edits to patient names don't seem to update on past prescriptions

### LINKED ISSUE #2236: Edits to patient names don't seem to update on past prescriptions
## What went wrong? 😲
 
If you edit a patient's name, for example Emily Porrello's here, which originally only had one `r`: 
image 

It doesn't update the name on prescriptions that were previously created for that patient. 
image

Should it?

## Expected behaviour 🤔
I think it should update, I suspect?

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Create a prescription for a patient 
2. Save the prescription 
3. Go to edit the patient and change the patient's name
4. Go back to the prescription and see whether it has been updated 

## Your environment 🌱

 - Version: 1.3.0
 - Platform:
   - [ ] android (tablet)
   - [x] browser (extra points if you tell us which one)
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2236

# 👩🏻‍💻 What does this PR do? 
The patient's full name wasn't being updated even though their first name and last name have, so have updated their full name too. 

# 🧪 How has/should this change been tested? 
- [ ] Create a prescription for a patient
- [ ] Go to patient view and change their first or last name
- [ ] Go back to prescriptions and see that their name has been changed

### CODE REVIEW

File: server/service/src/programs/patient/update_patient.rs [hunk 1]
```diff
@@ -46,11 +48,12 @@ fn generate(existing: NameRow, update: UpdatePatient) -> NameRow {
...
     NameRow {
         code,
-        first_name,
-        last_name,
+        first_name: first_name.clone(),
+        last_name: last_name.clone(),
         gender,
         date_of_birth,
         national_health_number: code_2,
+        name: patient_name(&first_name, &last_name),
```
Comment (clemens-msupply) [inline server/service/src/programs/patient/update_patient.rs:56]: You can avoid the clone by changing the order:
```rust
        name: patient_name(&first_name, &last_name),
        first_name: first_name,
        last_name: last_name,
```

File: server/service/src/programs/patient/update_patient.rs [hunk 2]
```diff
@@ -93,7 +96,7 @@ pub(crate) fn update_patient(
                 .rows
                 .pop()
                 .ok_or(UpdatePatientError::InternalError(
-                    "Can't find the just inserted patient".to_string(),
+                    "Can't find the newly created patient".to_string(),
```
Comment (clemens-msupply) [inline server/service/src/programs/patient/update_patient.rs:99]: Thanks!

**Review by clemens-msupply — CHANGES_REQUESTED** [review]
I think the service doesn't solve it nicely for `service/src/programs/patient/insert_patient.rs`. (sorry my fault)

Could you please create a service `InsertPatient` struct without a `name` field and then create the name field when generating the `NameRow` from the `InsertPatient` struct?

File: server/service/src/programs/patient/update_patient.rs (hunk 2 above)
Comment (roxy-dao) [reply server/service/src/programs/patient/update_patient.rs:99]: wrong message lol. I've changed it again! xD

**Comment by roxy-dao** [comment]
> I think the service doesn't solve it nicely for `service/src/programs/patient/insert_patient.rs`. (sorry my fault)
> 
> Could you please create a service `InsertPatient` struct without a `name` field and then create the name field when generating the `NameRow` from the `InsertPatient` struct?

[8a500f8](https://github.com/openmsupply/open-msupply/pull/2240/commits/8a500f87a8cf78c792758f9e59a875cc01c5945e) Thanks Clemens~

**Review by mark-prins — APPROVED** [review]
👍  it's a thing of beauty! working as described 😄

**Review by clemens-msupply — APPROVED** [review]
Thanks for the changes!

**Pull request merged**